packages/@aws-cdk/aws-ecs/lib/external/external-task-definition.ts

```diff
@@ -0,0 +1,83 @@
+import { Construct } from 'constructs';
+import { ImportedTaskDefinition } from '../../lib/base/_imported-task-definition';
+import {
+  CommonTaskDefinitionAttributes,
+  CommonTaskDefinitionProps,
+  Compatibility,
+  ITaskDefinition,
+  NetworkMode,
+  TaskDefinition,
+  Volume,
+} from '../base/task-definition';
+
+/**
+ * The properties for a task definition run on an External cluster.
+ */
+export interface ExternalTaskDefinitionProps extends CommonTaskDefinitionProps {
+
+}
+
+/**
+ * The interface of a task definition run on an External cluster.
+ */
+export interface IExternalTaskDefinition extends ITaskDefinition {
+
+}
+
+/**
+ * Attributes used to import an existing External task definition
+ */
+export interface ExternalTaskDefinitionAttributes extends CommonTaskDefinitionAttributes {
+
+}
+
+/**
+ * The details of a task definition run on an External cluster.
+ *
+ * @resource AWS::ECS::TaskDefinition
+ */
+export class ExternalTaskDefinition extends TaskDefinition implements IExternalTaskDefinition {
```

Comment:
Same issue here as with Service. We'd need some more validation I believe, namely on inferenceAccelerators. There may also be some additional ContainerDefinition fields that are not supported.

Reply:
Thanks. I have added more validations around this.

Reply:
Are the inference accelerator and volumes field the only unsupported task definition fields? How about the network mode, if someone tries to set `awsvpc` or `host`? What happens if an invalid network mode is passed in the constructor?

Reply:
Worked with @bvtujo on this and updated the default networkMode to `Bridge` for externalTasks. Other than this the customer cannot update or set the networkMode as its a read-only field.